frontend/src/containers/pages/AuthoringTool/CanvasSideBar/ComponentProperties/ButtonPropertiesComponent.jsx

```diff
@@ -25,6 +32,47 @@ export default function ButtonPropertiesComponent({
 }) {
   const { scenes, updateComponentProperty } = useContext(SceneContext);
 
+  const [newFlag, setNewFlag] = useState("");
+  const [open, setOpen] = useState(false);
+
+  const handleInputChange = (event) => {
+    setNewFlag(event.target.value);
+  };
+
+  const handleAddFlag = () => {
+    if (Object.keys(component.flagList).includes(`add ${newFlag}`)) {
+      setOpen(true);
+      return;
+    }
+
+    if (newFlag.trim() !== "") {
+      const newFlagList = {
+        ...component.flagList,
+        [`add ${newFlag}`]: false,
```

Comment:
Could we have two separate areas? As in - on the UI itself, instead of one list having `add FLAG_A` and `remove FLAG_A` in one dropdown, we have two dropdowns which represents the flags to add and remove respectively? Was there a reason for not going with the former?

Reply:
We could have two separate areas; I just felt like it was easier to just have one list so that everything is in one place.

I can change it to be two separate dropdown menus if you want me to do it that way.